odiglet/cmd/main.go

```diff
@@ -39,24 +41,14 @@ func odigletInitPhase() {
 	os.Exit(0)
 }
 
-func main() {
-	// If started in init mode
-	if len(os.Args) == 2 && os.Args[1] == "init" {
-		odigletInitPhase()
-	}
-
-	if err := log.Init(); err != nil {
-		panic(err)
-	}
-
-	log.Logger.V(0).Info("Starting odiglet")
-
-	// Load env
-	if err := env.Load(); err != nil {
-		log.Logger.Error(err, "Failed to load env")
-		os.Exit(1)
-	}
+type odiglet struct {
+	clientset     *kubernetes.Clientset
+	mgr           ctrl.Manager
+	ctx           context.Context
+	ebpfDirectors ebpf.DirectorsMap
+}
 
+func newOdiglet() *odiglet {
```

Comment:
Consider returning an error here and let the `main` call the logger and `os.Exit`